pkg/clicmd/prototype_use.go

```diff
@@ -84,34 +87,80 @@ ks prototype use single-port-deployment nginx-depl \
 `
 )
 
-func newGenerateCmd(a app.App) *cobra.Command {
-	cmd := newPrototypeUseCmd(a)
+func newGenerateCmd(fs afero.Fs) *cobra.Command {
+	cmd := newPrototypeUseCmd(fs)
 	cmd.Use = "generate <prototype-name> <component-name> [type] [parameter-flags]"
 
 	return cmd
 }
 
-func newPrototypeUseCmd(a app.App) *cobra.Command {
+func newPrototypeUseCmd(fs afero.Fs) *cobra.Command {
 	prototypeUseCmd := &cobra.Command{
-		Use:                "use <prototype-name> <componentName> [type] [parameter-flags]",
-		Short:              protoShortDesc["use"],
-		Long:               prototypeUseLong,
-		Example:            prototypeUseExample,
+		Use:     "use <prototype-name> <componentName> [type] [parameter-flags]",
+		Short:   protoShortDesc["use"],
+		Long:    prototypeUseLong,
+		Example: prototypeUseExample,
+		FParseErrWhitelist: cobra.FParseErrWhitelist{
+			UnknownFlags: true,
+		},
 		DisableFlagParsing: true,
-		RunE: func(cmd *cobra.Command, rawArgs []string) error {
-			if len(rawArgs) == 1 && (rawArgs[0] == "--help" || rawArgs[0] == "-h") {
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.DisableFlagParsing = false
+			err := cmd.ParseFlags(args)
+
+			if err != nil {
+				return cmd.FlagErrorFunc()(cmd, err)
+			}
+
+			if cmd.Flags().Lookup("help").Changed {
 				return cmd.Help()
 			}
 
+			args = removePersistentFlags(cmd.Flags(), args)
+
 			m := map[string]interface{}{
-				actions.OptionApp:       a,
-				actions.OptionArguments: rawArgs,
-				// We don't pass flagTLSSkipVerify because flag parsing is disabled
+				actions.OptionFs:        fs,
+				actions.OptionArguments: args,
 			}
+			addGlobalOptions(m)
 
 			return runAction(actionPrototypeUse, m)
 		},
 	}
 
 	return prototypeUseCmd
 }
+
+func removePersistentFlags(flags *pflag.FlagSet, args []string) []string {
```

Comment:
Can we utilize pflag to do the parsing? What about:
* One pass, parse with a flagset containing no flags, and `Unknown: true`. This should remove all flags and leave only positional arguments?
* Second pass, visit flags in original flagset. Construct new command line by appending all these with their values, plus the positional args.

Would that have the same outcome, without doing parsing logic on our end?

Reply:
^ reading that again, I think we would lose the unknown flags in the final output, which is not the intent.

Reply:
i'd really love not to copy pasta but i haven't found a good way out of it without modifying pflag